### PR TITLE
Add avro deserializer

### PR DESCRIPTION
**Pull request by dannyd4315**
This PR moves the avro deserializer from a 3rd party gem - https://github.com/karafka/avro (which is not up to date) to `Streamy` to give us more control.

It also updates the avro_turf gem to 0.10.0 to allow us to remove the `avro_patches` dependancy - https://github.com/salsify/avro-patches

This has been tested with `global-feeds` on `beta`

Once this has been updated then I will test and update streamy on `global-web`

### CODE REVIEW

File: lib/streamy/deserializers/avro_deserializer.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module Streamy
+  module Deserializers
+    class AvroDeserializer
+      require "avro_turf/messaging"
+
+      attr_reader :avro
```
Comment (karlentwistle) [inline lib/streamy/deserializers/avro_deserializer.rb]: Should this `attr_reader` be a private method 😄

Comment (dannyd4315) [reply lib/streamy/deserializers/avro_deserializer.rb]: Yep

File: test/avro_deserializer_test.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        "nowish"
+      end
+    end
+
+    def test_deserialized_message
+      SecureRandom.stubs(:uuid).returns("IAMUUID")
+
+      TestEvent.publish
+
+      assert_deserialized_message(
```
Comment (balvig) [inline test/avro_deserializer_test.rb]: Maybe feels a little early to extract a helper here?
I guess I would expect to see the class being exercised show up in the test block. Ie

```rb
module Streamy # see other tests
  # ...
  message = TestEvent.new.to_message
  result = Deserializers.AvroDeserializer.new.call(message)
  
  assert_equal {
    "type" => "test_event",
    # ...
  }, result # Note that "expected" comes first
```

Comment (dannyd4315) [reply test/avro_deserializer_test.rb]: 👍 Fair point

File: README.md [hunk 3]
```diff
@@ -183,6 +183,16 @@ class ReceivedPaymentHandler
 end
 ```
 
+Deserialization:
```
Comment (balvig) [inline README.md]: Nit:

```suggestion
### Deserialization
```

**Comment by dannyd4315** [comment]
🎉

**Pull request merged**